docs/api/merkle-api/03-examples.mdx

```diff
@@ -0,0 +1,202 @@
+---
+id: "examples"
+sidebar_position: 3
+title: "Examples"
+---
+
+import TabItem from "@theme/TabItem";
+import Tabs from "@theme/Tabs";
+
+# Examples of common flows
+
+Here are some common architectural flows you may need to integrate into your application. These include ways to gather
+the prerequisite data necessary for each backend [functionality](/api/merkle-api/functionality) described earlier.
+
+## Get a campaign's CID
+
+To get a hold of a campaign's IPFS CID you can:
+
+<ol type="A">
+  <li>
+    Check the [create](/contracts/v2/reference/periphery/interfaces/interface.ISablierV2MerkleStreamerFactory#Event)
+    event issued by the Merkle factory
+  </li>
+  <li>Use the Sablier V2 [subgraphs / indexers](/api/overview) to query for that particular piece of information.</li>
+</ol>
+
+For approach "B", run the following query against the official endpoints:
+
+<Tabs>
+  <TabItem value="TheGraph" label="The Graph" default>
+ ```GraphQL title:Example of a query for campaign data using The Graph's subgraphs
+    query getCampaignData($campaignId: String!){
+      campaign(id: $campaignId){
+        id
+        lockup
+        root
+        //highlight-start
+        ipfsCID
+        //highlight-end
+        aggregateAmount
+        totalRecipients
+      }
+    }
+```
+  </TabItem>
+  <TabItem value="Envio" label="Envio">
+ ```GraphQL title:Example of a query for campaign data using Envio's indexers
+    query getCampaignData($campaignId: String!){
+      Campaign(where: {id: {_eq: $campaignId}} ){
+        id
+        lockup
+        root
+        //highlight-start
+        ipfsCID
+        //highlight-end
+        aggregateAmount
+        totalRecipients
+      }
+    }
+```
+  </TabItem>
+</Tabs>
+
+## Check eligibility for an address
+
+To check if an address is eligible, you'll have to use the
+[/api/eligibility](/api/merkle-api/functionality#eligibility-apieligibility) route provided by the v2-merkle-api
+backend.
+
+#### Steps
+
+1. Get the campaign CID (see [the example](/api/merkle-api/examples#get-a-campaigns-cid) above)
+2. Call the`/api/eligibility` route using the CID and the wallet address
+
+Please read more in the dedicated route documentation within the
+[/api/eligibility](<(/api/merkle-api/functionality#eligibility-apieligibility)>) section of the functionality page.
+
+As an alternative, you can also check the eligibility of an address (and bypass the backend) by simply searching for
+that address in the list stored within the IPFS file from step 1. You'll still have to download the file first, using
+its CID.
+
+## Get the tokenId after a claim
+
+After someone claims, you may want to show them a preview of the stream (or its NFT). To do that, you'll have to get a
+hold of the `tokenId` (or `streamId`) related to that user's claim.
```

Comment:
shouldn't we prioritise `streamId` over `tokenId`? As in

```suggestion
hold of the `streamId` (or `tokenId`) related to that user's claim.
```

Reply:
I thought about this for a while, but I'm still not sure.

In terms of subgraphs and APIs, what is referred to as a "stream's id" is usually the `LL-1115511-12` formation. 

On one side `streamId` in those highlighted paragraphs would be following the definition as set in the contracts, but would suffer from that problem of having to define the source of this id well - is it the one in the contracts or the one in the app/subgraphs. 

On the other side, `tokenId` wouldn't have this problem. But it's harder to explain why the stream has a `tokenId` (to people not familiar with ERC721). Then again even the word "token" could introduce some problems.

Idk, maybe I'm overthinking this. Having some analysis paralysis at the moment. Which do you think it'd be best to use in this context?

Reply:
Good point. I have a question: in the graphql queries, do `$streamId` and `$streamIds` represent `tokenId` or `LL-1115511-12` formation? If its the latter, then I think the current approach works.

Reply:
It represents a formation (the one with the contract address in it): https://docs.sablier.com/api/subgraphs/architecture/structure#identifying

Reply:
Thank you for explaining. In that case, I think it's better to keep calling it `tokenId` to avoid any confusion between `Stream.id` and `streamId`.

> But it's harder to explain why the stream has a tokenId

We can expect developers using our API to already be familiar with the ERC721 standard. For them, `tokenId` should be self explanatory.